server/jobs/storage_backend.go

```diff
@@ -1,22 +1,96 @@
 package jobs
 
 import (
+	"fmt"
+	"io"
+
+	"github.com/graymeta/stow"
+	"github.com/pkg/errors"
 	"github.com/runatlantis/atlantis/server/core/config/valid"
+	"github.com/runatlantis/atlantis/server/logging"
 )
 
+const PageSize = 100
+
 //go:generate pegomock generate -m --use-experimental-model-gen --package mocks -o mocks/mock_storage_backend.go StorageBackend
 
 type StorageBackend interface {
 	// Read logs from the storage backend. Must close the reader
 	Read(key string) ([]string, error)
 
 	// Write logs to the storage backend
-	Write(key string, logs []string) (success bool, err error)
+	Write(key string, reader io.Reader, size int64) (bool, error)
+}
+
+type storageBackend struct {
+	location      stow.Location
+	logger        logging.SimpleLogging
+	containerName string
 }
 
-func NewStorageBackend(jobs valid.Jobs) StorageBackend {
-	// No storage backend configured, return Noop for now
-	return &NoopStorageBackend{}
+func (s *storageBackend) Read(key string) ([]string, error) {
+	return []string{}, nil
+}
+
+func (s *storageBackend) Write(key string, reader io.Reader, size int64) (bool, error) {
+	containerFound := false
+
+	// Function to write to container
+	writeFn := func(container stow.Container, err error) error {
+		if err != nil {
+			return errors.Wrapf(err, "walking containers at location: %s", s.location)
+		}
+
+		// Skip if not right container
+		if container.Name() != s.containerName {
+			return nil
+		}
+
+		containerFound = true
+		_, err = container.Put(key, reader, size, nil)
+		if err != nil {
+			s.logger.Warn("error uploading to %s", s.location, err)
+			return err
```

Comment:
If we are returning errors we should be returning wrapped errors so it is debuggable. Not sure if its worth logging here.

Reply:
Yup, makes sense! Thanks for the suggestion :)